### PR TITLE
Fixing must-gather ceph resources according to shell check

### PR DESCRIPTION
**Pull request by crombus**

### CODE REVIEW

**Comment by ashishranjan738** [comment]
@crombus I think this PR needs a rebase.

File: must-gather/collection-scripts/gather_ceph_resources [hunk 1]
```diff
@@ -7,18 +7,40 @@ if [ "${BASE_COLLECTION_PATH}" = "" ]; then
...
+CEPH_CONFIG_FILE="/etc/ceph/ceph.conf"
+CEPH_CONFIG_TEMPLATE="/templates/ceph.conf.template"
+CEPH_KEYRING_TEMPLATE="/templates/keyring.template"
 
 SED_DELIMITER=$(echo -en "\001");
 safe_replace () {
     sed "s${SED_DELIMITER}${1}${SED_DELIMITER}${2}${SED_DELIMITER}g"
 }
 
+generate_config() {
```
Comment (ashishranjan738) [inline must-gather/collection-scripts/gather_ceph_resources]: this func has been deleted

File: must-gather/collection-scripts/gather_ceph_resources [hunk 2]
```diff
@@ -7,18 +7,40 @@ if [ "${BASE_COLLECTION_PATH}" = "" ]; then
     BASE_COLLECTION_PATH=$(pwd)
 fi
 
-CEPH_COLLECTION_PATH="${BASE_COLLECTION_PATH}/ceph"
-POD_TEMPLATE="/templates/pod.template"
+CEPH_COLLLECTION_PATH="${BASE_COLLECTION_PATH}/ceph"
```
Comment (ashishranjan738) [inline must-gather/collection-scripts/gather_ceph_resources]: These constants are also not required anymore

File: must-gather/collection-scripts/gather_ceph_resources [hunk 3]
```diff
@@ -49,85 +71,54 @@ ceph_commands+=("ceph mgr dump")
...
 
 # Inspecting ceph related custom resources for all namespaces 
 for resource in "${ceph_resources[@]}"; do
-    oc adm --dest-dir=${CEPH_COLLECTION_PATH} inspect ${resource} --all-namespaces
+    oc adm --dest-dir="${CEPH_COLLLECTION_PATH}" inspect "${resource}" --all-namespaces
 done
 
 # Inspecting the namespace where ceph-cluster is installed
 for ns in $(oc get cephcluster --all-namespaces --no-headers | awk '{print $1}'); do
-    operatorImage=$(oc get pods -l app=rook-ceph-operator -n openshift-storage -o jsonpath="{range .items[*]}{@.spec.containers[0].image}+{end}" | tr "+" "\n" | head -n1)
```
Comment (ashishranjan738) [inline must-gather/collection-scripts/gather_ceph_resources]: we need this for collecting the output of non-running pods

File: must-gather/collection-scripts/gather_ceph_resources [hunk 4]
```diff
@@ -103,31 +101,31 @@ for ns in $(oc get cephcluster --all-namespaces --no-headers | awk '{print $1}')
...
     done
     
     # Collecting ceph prepare volume logs
     for node in $(oc get nodes -l cluster.ocs.openshift.io/openshift-storage='' --no-headers | grep -w 'Ready' | awk '{print $1}'); do
         printf "collecting prepare volume logs from node %s \n"  "${node}"
         NODE_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/namespaces/${ns}/osd_prepare_volume_logs/${node}
-        mkdir -p ${NODE_OUTPUT_DIR}
-        oc debug nodes/${node} -- bash -c "test -f /host/var/lib/rook/log/${ns}/ceph-volume.log && cat /host/var/lib/rook/log/${ns}/ceph-volume.log" > ${NODE_OUTPUT_DIR}/ceph-volume.log
+        mkdir -p "${NODE_OUTPUT_DIR}"
+        oc debug nodes/"${node}" -- bash -c "test -f /host/var/lib/rook/log/${ns}/ceph-volume.log && cat /host/var/lib/rook/log/${ns}/ceph-volume.log" > "${NODE_OUTPUT_DIR}"/ceph-volume.log
```
Comment (ashishranjan738) [inline must-gather/collection-scripts/gather_ceph_resources]: i think we should have a timeout here aswell.

Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: ok I will add it here also

**Comment by crombus** [comment]
https://github.com/openshift/ocs-operator/issues/235

**Pull request merged**